From c5fd69528315b7c7dbb437ad78b790f293455bd9 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Mon, 19 Jan 2015 08:38:21 -0500
Subject: [PATCH 042/315] xdmcp-display: drop extraneous code

This commit is like the previous commit but for xdmcp
displays instead of transient displays.

We set up vfuncs that just call into the default implementation.

This commit strips it.

https://bugzilla.gnome.org/show_bug.cgi?id=744764
---
 daemon/gdm-xdmcp-display.c | 66 ----------------------------------------------
 1 file changed, 66 deletions(-)

diff --git a/daemon/gdm-xdmcp-display.c b/daemon/gdm-xdmcp-display.c
index 41dd087f..9b929fae 100644
--- a/daemon/gdm-xdmcp-display.c
+++ b/daemon/gdm-xdmcp-display.c
@@ -61,7 +61,6 @@ enum {
 
 static void     gdm_xdmcp_display_class_init    (GdmXdmcpDisplayClass *klass);
 static void     gdm_xdmcp_display_init          (GdmXdmcpDisplay      *xdmcp_display);
-static void     gdm_xdmcp_display_finalize      (GObject              *object);
 
 G_DEFINE_ABSTRACT_TYPE (GdmXdmcpDisplay, gdm_xdmcp_display, GDM_TYPE_DISPLAY)
 
@@ -81,50 +80,6 @@ gdm_xdmcp_display_get_remote_address (GdmXdmcpDisplay *display)
         return display->priv->remote_address;
 }
 
-static gboolean
-gdm_xdmcp_display_create_authority (GdmDisplay *display)
-{
-        g_return_val_if_fail (GDM_IS_DISPLAY (display), FALSE);
-
-        return GDM_DISPLAY_CLASS (gdm_xdmcp_display_parent_class)->create_authority (display);
-}
-
-static gboolean
-gdm_xdmcp_display_add_user_authorization (GdmDisplay *display,
-                                          const char *username,
-                                          char      **filename,
-                                          GError    **error)
-{
-        return GDM_DISPLAY_CLASS (gdm_xdmcp_display_parent_class)->add_user_authorization (display, username, filename, error);
-}
-
-static gboolean
-gdm_xdmcp_display_remove_user_authorization (GdmDisplay *display,
-                                             const char *username,
-                                             GError    **error)
-{
-        return GDM_DISPLAY_CLASS (gdm_xdmcp_display_parent_class)->remove_user_authorization (display, username, error);
-}
-
-static gboolean
-gdm_xdmcp_display_manage (GdmDisplay *display)
-{
-        g_return_val_if_fail (GDM_IS_DISPLAY (display), FALSE);
-
-        GDM_DISPLAY_CLASS (gdm_xdmcp_display_parent_class)->manage (display);
-
-        return TRUE;
-}
-
-static gboolean
-gdm_xdmcp_display_unmanage (GdmDisplay *display)
-{
-        g_return_val_if_fail (GDM_IS_DISPLAY (display), FALSE);
-
-        GDM_DISPLAY_CLASS (gdm_xdmcp_display_parent_class)->unmanage (display);
-        return TRUE;
-}
-
 static void
 _gdm_xdmcp_display_set_remote_address (GdmXdmcpDisplay *display,
                                        GdmAddress      *address)
@@ -212,13 +167,7 @@ gdm_xdmcp_display_class_init (GdmXdmcpDisplayClass *klass)
 
         object_class->get_property = gdm_xdmcp_display_get_property;
         object_class->set_property = gdm_xdmcp_display_set_property;
-        object_class->finalize = gdm_xdmcp_display_finalize;
 
-        display_class->create_authority = gdm_xdmcp_display_create_authority;
-        display_class->add_user_authorization = gdm_xdmcp_display_add_user_authorization;
-        display_class->remove_user_authorization = gdm_xdmcp_display_remove_user_authorization;
-        display_class->manage = gdm_xdmcp_display_manage;
-        display_class->unmanage = gdm_xdmcp_display_unmanage;
         display_class->get_timed_login_details = gdm_xdmcp_display_get_timed_login_details;
 
         g_type_class_add_private (klass, sizeof (GdmXdmcpDisplayPrivate));
@@ -249,18 +198,3 @@ gdm_xdmcp_display_init (GdmXdmcpDisplay *xdmcp_display)
 
         xdmcp_display->priv = GDM_XDMCP_DISPLAY_GET_PRIVATE (xdmcp_display);
 }
-
-static void
-gdm_xdmcp_display_finalize (GObject *object)
-{
-        GdmXdmcpDisplay *xdmcp_display;
-
-        g_return_if_fail (object != NULL);
-        g_return_if_fail (GDM_IS_XDMCP_DISPLAY (object));
-
-        xdmcp_display = GDM_XDMCP_DISPLAY (object);
-
-        g_return_if_fail (xdmcp_display->priv != NULL);
-
-        G_OBJECT_CLASS (gdm_xdmcp_display_parent_class)->finalize (object);
-}
-- 
2.15.1

